server/src/main/java/hu/flowacademy/epsilon/sport_event_organizer/service/MatchService.java

```diff
@@ -0,0 +1,261 @@
+package hu.flowacademy.epsilon.sport_event_organizer.service;
+
+import hu.flowacademy.epsilon.sport_event_organizer.model.Cup;
+import hu.flowacademy.epsilon.sport_event_organizer.model.Match;
+import hu.flowacademy.epsilon.sport_event_organizer.model.Team;
+import hu.flowacademy.epsilon.sport_event_organizer.repository.MatchRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@Transactional
+@Slf4j
+public class MatchService {
+
+    @Autowired
+    private CupService cupService;
+
+    @Autowired
+    private TeamService teamService;
+
+    @Autowired
+    private MatchRepository matchRepository;
+
+    public List<Match> generateGroupMatches(String cupName) {
+        Cup cup = cupService.getByName(cupName);
+        List<Team> cupTeams = new ArrayList<Team>(cupService.getApprovedTeams(cupName));
+        List<Match> matches = new ArrayList();
+        if (cupTeams.size() == 8) {
+            String groupName = "Group1";
+            for (int i = 0; i < 2; i++) {
+                for (int j = 1; j < 4; j++) {
+                    Match match = new Match();
+                    match.setTeamA(cupTeams.get(0));
```

Comment:
A 3 for törzse ugyanazt csinálja. Szervezzétek ki egy függvénybe. `createMatch` lehet a neve